### PR TITLE
Get workspace from project.projectDir.

### PR DESCRIPTION
**Pull request by ecavinat**

### CODE REVIEW

File: scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java [hunk 1]
```diff
@@ -156,21 +160,14 @@ private synchronized String buildCurrentExecutionPath() {
...
-			
+			String fileName = this.getConfig().getFileName().toString();
 			if(fileName.endsWith(".conf")) {
 				fileName = fileName.replace(".conf", "");
 			}
 			if(fileName.endsWith(".properties")) {
 				fileName = fileName.replace(".properties", "");
 			}
-			String module = getModule().getFileName().toString();
+			final String module = workspace.relativize(this.getModule().toAbsolutePath()).toString().replace(Configuration.MODULES_FOLDER + File.separator, "");
```
Comment (fbrubbo) [inline scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java]: Why do you need to all this? Isn't it the enough to get the to getModule().getFileName() as the previous code? Looks like we have added an unneeded complexity.

**Review by fernandorubbo — COMMENTED** [review]
Please, review my comments

File: scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java [hunk 2]
```diff
@@ -156,21 +160,14 @@ private synchronized String buildCurrentExecutionPath() {
...
 			Path workspace = workspace();
-			Path config = getConfig();
-			String fileName;
-			if(config!=null) {
-				fileName = workspace.relativize(config).toString().replace("config/", "");
-			} else {
-				fileName = workspace.toString().replace("config/", "");
-			}
-			
+			String fileName = this.getConfig().getFileName().toString();
```
Comment (ecavinat) [inline scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java]: For me, a configuration file must be setted. Maybe a exception should be thrown if getConfig had a null value...

File: scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java (hunk 1 above)
Comment (ecavinat) [inline scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java]: I used multi-level module folders and I need a better content visualization (for files and folders) for the folders "log" and "backup" , which reflect that modules configuration. If I use only getModule().getFileName() for that, the contents of "log" and "backup" folders didn't reflect those organization (modules subfolders). For exemple, I used this organization:

modules
- systemA
-- moduleA
-- moduleB
-- moduleC
- systemB
-- moduleD
- systemC
-- moduleE

log
- systemA
-- moduleA
--- file.log
-- moduleB
--- file.log
-- moduleC
--- file.log
- systemB
-- moduleD
--- file.log
- systemC
-- moduleE
--- file.log

backup
- systemA
-- moduleA
--- installed files
-- moduleB
--- installed files
-- moduleC
--- installed files
- systemB
-- moduleD
--- installed files
- systemC
-- moduleE
--- installed files

Without my code, the log and backup folders make sense only if I put the modules as directly folders of "modules" folder.

Comment (fbrubbo) [inline scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java]: I never thought of using multi-level modules. Does it work? 
Have you tried these new changes on the linux server or only on the windows? Have you tried it inside and outside of eclipse?

File: scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java (hunk 2 above)
Comment (fbrubbo) [inline scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java]: That is ok for me. if you make sure there will be no cases getConfig() returns null.

Comment (ecavinat) [inline scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java]: I add a new change to thrown a exception if getConfig is null.

File: scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java (hunk 1 above)
Comment (ecavinat) [inline scd4j/src/main/java/com/datamaio/scd4j/conf/Configuration.java]: I tested from Eclipse and through command line, in Windows and Linux. It works.

**Pull request merged**